database2/cassandraModel.js

```diff
@@ -0,0 +1,147 @@
+    // 'kitchen': 2010001 - 4020000,
+    return [2010001, 4020000];
+  } else {
+    //'living-room': 1 - 2010000,
+    return [1, 2010000];
+  }
+};
+
+const getCatId = (category) => {
+  if (category === 'laundry') {
+    return 7;
+  } else if (category === 'closet') {
+    return 6;
+  } else if (category === 'bathroom') {
+    return 5;
+  } else if (category === 'bedroom') {
+    return 4;
+  } else if (category === 'dining') {
+    return 3;
+  } else if (category === 'kitchen') {
+    return 2;
+  } else {
+    return 1;
+  }
+};
+
+client.connect();
+module.exports = {
+  getOne : (request, response) => {
+    console.log("get params received in model: ", request.params);
+    const productId = request.params.id;
+    //const productId = 7;
+    const query = `SELECT * FROM products WHERE productid = ?`;
+    client.execute(query, [productId], {prepare: true})
+    .then(result => {
+      const row = result.rows[0];
+      console.log("price result", row.price)
+      const low = parseInt(row.price) - 50;
+      const high = parseInt(row.price) + 50;
+      const values = getRange(productId, null).concat(low, high);
+      const query = `SELECT * FROM products WHERE productid >= ? AND productid <= ? AND price >= ? AND price <= ? limit 8 allow filtering`;
+      client.execute(query, values, {prepare: true})
+      .then(result => {
+        console.log(result.rows, "<---similar products");
+        response.status(200).send(result.rows)
+      })
+    })
+    .catch(err => {
+      console.error(err.stack);
+      response.status(404).end('product not found')
+    })
+  },
+
+  create : (request, response) => {
+    console.log("post body received in model: ", request.body)
+    const b = request.body;
+    const idQuery = `SELECT productid FROM maxids WHERE category = ? allow filtering`
+    client.execute(idQuery, [b.category], {prepare: true})
+    .then(result => {
+      const rows = result.first();
+      const productId = rows.productid + 1;
+      const catId = getCatId(b.category);
+      const updateQuery = `UPDATE maxids SET productid = ? WHERE id = ?`
+      const insertQuery = `INSERT INTO products (productid, name, category, price, rating, imageurl, onsale) VALUES (?, ?, ?, ?, ?, ?, ?)`;
+      const queries = [
+        {query: updateQuery, params:[productId, catId]},
+        {query: insertQuery, params: [productId, b.name, b.category, b.price, b.rating, b.imageUrl, b.onSale]}
+      ]
+      return client.batch(queries, {prepare: true})
+    })
+    .then((result) => {
+      console.log("created new product and updated maxIds");
+      response.sendStatus(201)
+    })
+    .catch(err => {
+      console.error("create query failed: ", err.stack);
+      response.sendStatus(400)
+    })
+  },
+
+  remove : (request, response) => {
+    console.log("delete request params received in model: ", request.params);
+    const query = `DELETE FROM products WHERE productid = ?`
+    const productId = request.params.id;
```

Comment:
Same thing I said at the beginning of your create function. You only ever have the one productId and you're already using a string template so you could cut down a line if you want to.

Reply:
I might since it makes it easier to read/looks nicer and doesn't seem to actually cause any problems :)